src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,1246 @@
+        t = torch.from_numpy(t).to(depth).view(E, 1, 1, 1)
+        out = depth * s + t
+        return out
+
+    def ensemble(
+        depth_aligned: torch.FloatTensor, return_uncertainty: bool = False
+    ) -> Tuple[torch.FloatTensor, Optional[torch.FloatTensor]]:
+        uncertainty = None
+        if reduction == "mean":
+            prediction = torch.mean(depth_aligned, dim=0, keepdim=True)
+            if return_uncertainty:
+                uncertainty = torch.std(depth_aligned, dim=0, keepdim=True)
+        elif reduction == "median":
+            prediction = torch.median(depth_aligned, dim=0, keepdim=True).values
+            if return_uncertainty:
+                uncertainty = torch.median(torch.abs(depth_aligned - prediction), dim=0, keepdim=True).values
+        else:
+            assert False
+        return prediction, uncertainty
+
+    def cost_fn(st: np.ndarray) -> float:
+        cost = 0.0
+        s, t = np.split(st, 2)
+        depth_aligned = align_depth(depth_to_align, s, t)
+
+        for i, j in torch.combinations(torch.arange(E)):
+            diff = depth_aligned[i] - depth_aligned[j]
+            cost += (diff**2).mean().sqrt().item()
+
+        if regularizer_strength > 0:
+            prediction, _ = ensemble(depth_aligned, return_uncertainty=False)
+            err_near = (0.0 - prediction.min()).abs().item()
+            err_far = (1.0 - prediction.max()).abs().item()
+            cost += (err_near + err_far) * regularizer_strength
+
+        return cost
+
+    init_min = depth_to_align.reshape(E, -1).min(dim=1).values
+    init_max = depth_to_align.reshape(E, -1).max(dim=1).values
+    init_s = 1.0 / (init_max - init_min).clamp(min=1e-6)
+    init_t = -init_s * init_min
+    init_st = torch.cat((init_s, init_t)).cpu().numpy()
+
+    res = minimize(
+        cost_fn,
+        init_st,
+        method="BFGS",
+        tol=tol,
+        options={"maxiter": max_iter, "disp": False},
+    )
+    s, t = np.split(res.x, 2)
+
+    depth = align_depth(depth, s, t)
+    depth, uncertainty = ensemble(depth, return_uncertainty=output_uncertainty)
+
+    depth_min = depth.min()
+    depth_max = depth.max()
+    depth_range = (depth_max - depth_min).clamp(min=1e-6)
+    depth = (depth - depth_min) / depth_range
+    if output_uncertainty:
+        uncertainty /= depth_range
+
+    return depth, uncertainty  # [1,1,H,W], [1,1,H,W]
+
+
+def colormap(
+    image: Union[np.ndarray, torch.Tensor],
+    cmap: str = "Spectral",
+    bytes: bool = False,
+    force_method: Optional[str] = None,
+) -> Union[np.ndarray, torch.Tensor]:
+    assert force_method in (None, "matplotlib", "custom")
+    if not (torch.is_tensor(image) or isinstance(image, np.ndarray)):
+        raise ValueError("Argument must be a numpy array or torch tensor.")
+
+    def method_matplotlib(image, cmap, bytes=False):
+        try:
+            import matplotlib
+        except ImportError:
+            return None
```

Comment:
Is `matplotlib` a required dependency here? If so, we need to tackle its import with `is_matplotlib_available()`. Here's an example:

https://github.com/huggingface/diffusers/blob/0d7c4790235ac00b4524b492bc2a680dcc5cf6b0/src/diffusers/utils/import_utils.py#L315

Reply:
We use matplotlib to visualize the depth map with named [colormaps](https://matplotlib.org/stable/users/explain/colors/colormaps.html). I also provided a fallback for default arguments when it is unavailable.

Reply:
Fixed; visualization functions are also moved to export utils. matplotlib is treated with `is_matplotlib_available`, same logic is applied in the depth pipeline to handle scipy.